telegram/ext/filters.py

```diff
@@ -603,6 +603,41 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+
+            Example:
+                ``Filters.document.file_extension('jpg')`` filters files with extension ``.jpg``.
+            """
+
+            def __init__(self, file_extension: Optional[str]):
+                """Initialize the extension you want to filter.
+
+                Args:
+                    file_extension (str, optional): media file extension you want to filter.
+                """
+                if file_extension is None:
+                    self.file_extension = file_extension
+                elif file_extension.startswith("."):
+                    self.file_extension = file_extension
+                else:
+                    self.file_extension = f".{file_extension}"
+                self.name = f"Filters.document.file_extension({self.file_extension!r})"
```

Comment:
why is the `!r` necessary?

Reply:
Because it should differ for `None` and `"None"`.

Reply:
but `str(None) == repr(None)`? o.0 shouldn't it rather be `f'{self.f_e if self.f_e else ""}'`, then?

Reply:
`str("None") != repr("None")` — that's what i mean. Please read the test to understand my intentions. May be i just got the task wrong. In short, `None` and `""` behaviors differ: `""` means filenames ending with a dot and `None` means empty filter that always returns `True` (except if there's no filename at all).

Reply:
I do understand that `'None'` and `None` are not the same as input. Still I find `file_extension(None)` resulting in `'Filters.doucment.file_extension(None)'` a bis misleading as name (the difference to `"file_extension('None')"` being marginal and would prefer `'Filters.doucment.file_extension()'`. Anyway, I'm not too passionate about that, but it would maybe be good to explicitly ad tests checking that the name is set correctly

Reply:
Should we accept `None` at all? Argument is not optional, so `None` can only be passed explicitly. Adding `file_extension()` to the name may be misleading.

Reply:
Made it optional and removed `None` from name. But behavior question is still open. I've separated all `None`-related checks in one test, please read.

Reply:
No, I think `None` should have to be passed explicitly. Though thinking about it, it would actually make sense to not just return `True` in that case but on the contrary, check that the file has *no* file extension, i.e. `return '.' not in file_extension`. Ofc that should be documented. 
In that case I do agree that `Filters.f_e(None)` would be a more appropriate naming.
I'll try to review this and the types PR later today